### ISSUE TITLE
Raise warning when one or more parameters has no data

### ISSUE THREAD

**Issue by volcan01010** [BritishGeologicalSurvey/pyvolcans#4]
### Summary

As a pyvolcans user, I want a warning when one or more parameters has no data so that I can exclude it from the analysis and get fairer comparisons.

### Description

The results for Auckland Field look like this:

```
$ pyvolcans -v "Auckland Field"
PyVOLCANS: Supplied weights: {'tectonic_setting': 0.2, 'geochemistry': 0.2, 'morphology': 0.2, 'eruption_size': 0.2, 'eruption_style': 0.2}
Top 10 analogue volcanoes for Auckland Field, New Zealand (241020):
                    name         country  smithsonian_id  total_analogy  ATs       AG  AM  ASz      ASt
Pali-Aike Volcanic Field Chile-Argentina          358080       0.760784  0.2 0.160784 0.2  0.0 0.200000
                  Huambo            Peru          354005       0.756863  0.2 0.156863 0.2  0.0 0.200000
                Pinacate          Mexico          341001       0.752778  0.2 0.177778 0.2  0.0 0.175000
     Cayutue-La Vigueria           Chile          358012       0.749020  0.2 0.149020 0.2  0.0 0.200000
       Andahua-Orcopampa            Peru          354004       0.749020  0.2 0.149020 0.2  0.0 0.200000
           Chichinautzin          Mexico          341080       0.745384  0.2 0.151634 0.2  0.0 0.193750
 Naolinco Volcanic Field          Mexico          341095       0.744608  0.2 0.169608 0.2  0.0 0.175000
  Kaikohe-Bay of Islands     New Zealand          241010       0.741176  0.2 0.141176 0.2  0.0 0.200000
    Michoacan-Guanajuato          Mexico          341060       0.733111  0.2 0.149020 0.2  0.0 0.184091
San Pablo Volcanic Field     Philippines          273060       0.725980  0.2 0.150980 0.2  0.0 0.175000
```

Note that the ASz column is all 0, suggesting that there is no source data for eruption size.  This caps the total analogy value at 0.8, which suggests there are no volcanoes with a very good match.  It would be good to raise a Warning to let users know about this.  (This could be similar to the warning when many volcanoes have the same score).

The warning would be especially valuable when users are not using verbose mode as they wouldn't see the individual scores.

See this Tweet thread for details: https://twitter.com/Tyranakisaurus/status/1384469161569566720

### Acceptance criteria

+ [x] `pyvolcans "Auckland Field"` raises a Warning that there is no eruption_size data
+ [x] Warning recommends manually setting weights to exclude bad parameter

**Comment by PTierz** [comment]
Hi @volcan01010!

Thank you very much for setting up this issue!

I have tried to address the issue in commits d4dda82, 753fe80, inside the `plot-bars` branch. I am checking the acceptance criteria above because they are working as expected in my local machine, but please double-check these additional acceptance criteria below:

+ [x] `pyvolcans -v 221251` raises the following warning message: `PyVOLCANS: WARNING!!! The following volcanological criteria do not have any data available for the selected target volcano (Unnamed): morphology, eruption_size, eruption_style. Please consider excluding these criteria from your weighting scheme (i.e. setting their weights to zero).`
+ [x] `pyvolcans -v 243091` raises the following warning message: `PyVOLCANS: WARNING!!! The following volcanological criteria do not have any data available for the selected target volcano (Unnamed): morphology. Please consider excluding these criteria from your weighting scheme (i.e. setting their weights to zero).`
+ [x] `pyvolcans -v 274050` raises the following warning message: `PyVOLCANS: WARNING!!! The following volcanological criteria do not have any data available for the selected target volcano (Unnamed): geochemistry, morphology, eruption_style. Please consider excluding these criteria from your weighting scheme (i.e. setting their weights to zero).`

Very many thanks in advance!
Pablo

PS. Please note that there were some issues with the tests when introducing the `my_volcano_data_dictionary` that checks for data availability for each of the volcanological criteria, and considering the selected target volcano. I have implemented some changes to make the tests pass again (please see commit c54c715), but there might be better, more stable solutions to the one that I have chosen. Thank you very much!

**Comment by volcan01010** [comment]
I just ran this and all the acceptance criteria pass.  Nice one.

Looking at the code, there are a couple of changes to make:

+ [x] Replace the PyvolcansError with a warnings.warn (https://pymotw.com/2/warnings/).  Then you don't need to catch it and pass them.
+ [x] Move the checking of the criteria/weighting results into the `calculate_weighted_analogy_matrix`.  This way you don't need to return the `my_volcano_data_dictionary` back up to the main function.

**Comment by jifarquharson** [comment]
### Summary ###
`pyVOLCANS` documentation could do with a note on GVP naming conventions.

### Description ###
Many volcanoes have names that are more than one word, e.g. Rincon de la Vieja (Costa Rica) or St. Helens (USA): as it stands, if a user were to enter `$ pyvolcans Rincon de la Vieja` (consistent with the examples given, i.e. `$ pyvolcans Hekla`), `pyvolcans` will throw an `"unrecognized argument"` error. This can be avoided simply by wrapping the multi-word string in quotes, but as far as I can tell this isn't mentioned in the documentation. 

On a related note, volcano names crawled from the Smithsonian's GVP are often in a {Name, Prefix} format, which might not be immediately obvious for a lay end-user. A common example is "Fournaise, Piton de la." It seems that the fuzzy matching does a reasonable job of catching these in most cases; for example, "Fournaise, Piton de la" is the 9th alternative option given by `fuzzy_matching()` for "Piton de la Fournaise," and "Sawad, Harra Es-" is the 4th option given for "Harra Es-Sawad," and so on. Other systems don't fare as well (e.g. Cerro Bravo or Jabal Yar), and don't offer the correct option. While arbitrarily increasing the fuzzy limit, e.g. by including
`limit = 10 if len(volcano_name.split()) == 1 else 20`
inside the `fuzzy_matching` function helps to a degree, this workaround still doesn't capture a dozen or so systems (for example, "Bravo, Cerro" is the 42nd-best fuzzy match for "Cerro Bravo"). While writing a targeted workaround for this is probably overkill given that it is potentially a niche problem, it would be useful to have some note in the documentation regarding how to input names, the format they should be in, and where to look if your volcano appears not to be on the list, e.g. due to formatting, punctuation, or usage of a different name (one can imagine researchers looking for "Fagradalsfjall," for example, rather than "Krýsuvík-Trölladyngja"). 

### Acceptance criteria ###

- [ ] Updated documentation/instructions

Ping [3649](https://github.com/openjournals/joss-reviews/issues/3649)

**Comment by PTierz** [comment]
This Issue was solved through changes implemented in the `fix-open-issue` branch, which has now been merged into `main` (see pull request #13). I am, therefore, closing the Issue with this comment.

If there was any outstanding task to be implemented, please let us know: @jifarquharson

Very many thanks in advance.
Best regards,
Pablo
